docs/source/new-docs/tutorials/2024-tutorial.rst

```diff
@@ -0,0 +1,515 @@
+YAML file. By product we mean development of a new functionality using GeoIPS as a base.
+The type (Python module / YMAL) of the product plugin is determined by its interface.
+
+Plugins are stored in installable Python packages that register their payload with
+GeoIPS through the use of
+`entrypoints <https://packaging.python.org/en/latest/specifications/entry-points/>`_.
+
+#. Module-based Plugin
+
+   A module-based plugin is a plugin that extends GeoIPS by adding new
+   functionality that is capable of performing an action (e.g. apply an algorithm,
+   read data, apply formatting, etc.).  Module-based plugins are defined as a
+   single python module that contains a handful of required top-level variables and
+   a single function that performs the action of the plugin. Examples of
+   module-based plugins include ``algorithms``, ``readers``, and various types of
+   formatters.
+
+#. YAML-based Plugin
+
+   A YAML-based plugin is a plugin that extends GeoIPS by adding a new set of
+   static configuration options for GeoIPS.  Examples of YAML-based plugins include
+   ``sectors``, ``products``, and ``feature-annotators``.
+
+.. _required-attributes:
+
+Plugin Attributes:
+------------------
+
+The following are the top level attributes required while defining a new product plugin:
+
+#. Interface
+
+   An ``interface`` defines a class of GeoIPS plugins that extend the same type of
+   functionality within GeoIPS. For example, some commonly used interfaces include the
+   ``algorithms``, ``colormappers``, and ``sectors`` interfaces.
+
+#. Family
+
+   A ``family`` is a subset of an interface's plugins which accept specific sets of
+   arguments/properties. Module-based plugins of the same ``family`` have similar call
+   signatures. YAML-based plugins of the same ``family`` are validated against the same
+   schema (i.e. they contain the same properties).
+
+#. Docstring
+
+   A ``docstring`` is a chunk of documentation which describes what your plugin does. This
+   property is required for every GeoIPS plugin created, module-based or YAML-based. We
+   require this property for proper documentation of created plugins, and it will also be
+   a useful feature later on when the GeoIPS Command Line Interface (CLI) is created, as
+   you will be able to see what each plugin does provided the ``docstring`` for that plugin
+   is filled.
+
+.. _plugin-development-setup:
+
+Product Plugin Development Initial Setup
+****************************************
+
+Before creating a new product for CLAVR-x Cloud-Top-Height, let's get the initial setup done:
+
+#. To develop a new GeoIPS plugin, install :ref:`GeoIPS<linux-installation>` and make sure that
+   you have ``geoips`` Python environment enabled throughout this tutorial using
+
+   .. code-block:: shell
+
+    mamba activate geoips # activating python envrionment
+
+   You will know if geoips environment is enabled if it shows up ahead of your username in your command prompt.
```

Comment:
Please edit this for brevity, and use active language. For example: "You will know the geoips conda environment is enabled if it appears before your username in the command prompt."